api/v1alpha1/proxmoxmachine_types.go

```diff
@@ -145,6 +145,7 @@ const (
 )
 
 // VirtualMachineCloneSpec is information used to clone a virtual machine.
+// TODO +kubebuilder:validation:XValidation:rule="(self.sourceNode != ” || self.templateID != 0) && (self.templateSelector != null)",message="must define either SourceNode and TemplateID, OR TemplateSelector"
```

Comment:
Make sourceNode a pointer and check for it to be null.
I used that trick here: https://github.com/ionos-cloud/cluster-api-provider-proxmox/blob/main/api/v1alpha1/proxmoxmachine_types.go#L363